src/components/FlowChart/FlowSelection.jsx

```diff
@@ -1,56 +1,118 @@
+    <i className="material-icons text-white ml-1 mr-2">wifi</i>
+  ) : (
+    <i className="material-icons text-warning ml-1 mr-2">wifi_off</i>
+  );
+}
+
+function DeleteFlowButton({ show, deleteFlow }) {
+  if (!show) return null;
+  return (
+    <i className="material-icons text-danger float-right" onClick={deleteFlow}>
+      delete
+    </i>
+  );
+}
+
+function EditFlowsButton({ isEditing, onClick }) {
+  return isEditing ? (
+    <i
+      onClick={onClick}
+      className="material-icons float-right cursor-pointer text-success"
+    >
+      done
+    </i>
+  ) : (
+    <i onClick={onClick} className="material-icons float-right cursor-pointer">
+      edit
+    </i>
+  );
+}
+
 export default function FlowSelection({
   flowOptions,
   loadFlow,
   selectedFlowId,
   createNewFlow,
+  deleteFlow,
   connected,
 }) {
+  const [isEditing, setIsEditing] = useState(false);
+
+  const toggleEditing = () => {
+    setIsEditing((currentIsEditing) => !currentIsEditing);
+  };
+
+  const userFlows = Object.entries(flowOptions).filter(
+    ([id, flow]) => flow.type !== "example"
+  );
+
+  const exampleFlows = Object.entries(flowOptions).filter(
+    ([id, flow]) => flow.type === "example"
+  );
+
+  const currentFlow = flowOptions[selectedFlowId];
+
   return (
     <Dropdown className="flow-selection">
-      <Dropdown.Toggle>{flowOptions[selectedFlowId].name}</Dropdown.Toggle>
+      <Dropdown.Toggle>
+        {currentFlow.name}
+        <TitleConnectionIndicator
+          show={currentFlow.type === "remote"}
+          connected={connected}
+        />
+      </Dropdown.Toggle>
       <Dropdown.Menu>
         <Dropdown.Item onClick={createNewFlow}>
           <i className="material-icons">add</i>New Flow
         </Dropdown.Item>
-        <Dropdown.Header className="text-uppercase">Your Flows</Dropdown.Header>
-        {Object.entries(flowOptions)
-          .filter(([id, flow]) => flow.type !== "example")
-          .map(([flowId, flow], idx) => (
-            <Dropdown.Item onClick={() => loadFlow(flowId)} key={idx}>
-              {selectedFlowId === flowId ? (
-                <i className="material-icons text-primary">
-                  radio_button_checked
-                </i>
-              ) : (
-                <i className="material-icons">radio_button_unchecked</i>
-              )}
-              {flow.name}
-              {flow.type === "remote" &&
-                (connected ? (
-                  <i className="material-icons ml-2 text-success">wifi</i>
-                ) : (
-                  <i className="material-icons ml-2 text-warning">wifi_off</i>
-                ))}
-            </Dropdown.Item>
-          ))}
+        <Dropdown.Header className="text-uppercase">
+          Your Flows{" "}
+          <EditFlowsButton isEditing={isEditing} onClick={toggleEditing} />
+        </Dropdown.Header>
+        {userFlows.map(([flowId, flow], idx) => (
+          <Dropdown.Item onClick={() => loadFlow(flowId)} key={idx}>
```

Comment:
It's not always a problem to use the index as a key but it can be a problem if the number of things changes between renders and the position/index does not identify the rendered item.
`flowId` is unique and we already have it, that's what I would use.